src/Filesystem/Provider/FileAccess.php

```diff
@@ -0,0 +1,257 @@
+	 *
+	 * @param string $path                  The path to the file which should be read.
+	 *
+	 * @return string   The file content.
+	 *
+	 * @throws FileNotFoundException        If the file doesn't exist.
+	 * @throws IOException                  If the file could not be red.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function read($path);
+
+
+	/**
+	 * Checks whether a file exists.
+	 *
+	 * @param string $path The file path which should be checked.
+	 *
+	 * @return bool True if the file exists, otherwise false.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function has($path);
+
+
+	/**
+	 * Deletes a file.
+	 *
+	 * @param string $path              The path to the file which should be deleted.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException    If the file was not found.
+	 * @throws IOException              If the file was found but the delete operation finished with errors.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function delete($path);
+
+
+	/**
+	 * Reads the entire file content into a string and removes the file afterwards.
+	 *
+	 * @param string $path  The file which should be red and removed.
+	 *
+	 * @return string       The entire file content.
+	 *
+	 * @throws FileNotFoundException    If the file was not found.
+	 * @throws IOException              If the file could not red or deleted.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function readAndDelete($path);
+
+
+	/**
+	 * Moves a file from the source to the destination.
+	 *
+	 * @param string $path      The current path of the file which should be moved.
+	 * @param string $newPath   The new path of the file.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException        If the source file is not found.
+	 * @throws FileAlreadyExistsException   If the destination file is already existing.
+	 * @throws IOException                  If the file could not be moved.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function rename($path, $newPath);
+
+
+	/**
+	 * Copy the source file to a destination.
+	 *
+	 * @param string $path      The source path to the file which should be copied.
+	 * @param string $copyPath  The destination path of the file copy.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException        If the source file does not exist.
+	 * @throws FileAlreadyExistsException   If the destination file already exists.
+	 * @throws IOException                  If the file could not be copied to the destination.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function copy($path, $copyPath);
+
+
+	/**
+	 * Get a files mime-type.
+	 *
+	 * @param string $path  The file which should be used to get the mime-type.
+	 *
+	 * @return string   The mime-type of the file.
+	 *
+	 * @throws FileNotFoundException    If the file is not found.
+	 * @throws IOException              If the mime-type could not be determined.
+	 */
+	public function getMimeType($path);
```

Comment:
I think the mimetype is currently the most used metadata of files (after the filesize) in ILIAS. Having this explicitly is nice.